tokio-trace/src/lib.rs

```diff
@@ -0,0 +1,760 @@
+//! as long as the longest-executing span in its subtree.
+//!
+//! In addition, data may be associated with spans. A span may have _fields_ ---
+//! a set of key-value pairs describing the state of the program during that
+//! span; an optional name, and metadata describing the source code location
+//! where the span was originally entered.
+//!
+//! ## Events
+//!
+//! An [`Event`] represents a _point_ in time. It signifies something that
+//! happened while the trace was executing. `Event`s are comparable to the log
+//! records emitted by unstructured logging code, but unlike a typical log line,
+//! an `Event` may occur within the context of a `Span`. Like a `Span`, it
+//! may have fields, and implicitly inherits any of the fields present on its
+//! parent span, and it may be linked with one or more additional
+//! spans that are not its parent; in this case, the event is said to _follow
+//! from_ those spans.
+//!
+//! Essentially, `Event`s exist to bridge the gap between traditional
+//! unstructured logging and span-based tracing. Similar to log records, they
+//! may be recorded at a number of levels, and can have unstructured,
+//! human-readable messages; however, they also carry key-value data and exist
+//! within the context of the tree of spans that comprise a trase. Thus,
+//! individual log record-like events can be pinpointed not only in time, but
+//! in the logical execution flow of the system.
+//!
+//! Events are represented as a special case of spans --- they are created, they
+//! may have fields added, and then they close immediately, without being
+//! entered.
+//!
+//! ## `Subscriber`s
+//!
+//! As `Span`s and `Event`s occur, they are recorded or aggregated by
+//! implementations of the [`Subscriber`] trait. `Subscriber`s are notified
+//! when an `Event` takes place and when a `Span` is entered or exited. These
+//! notifications are represented by the following `Subscriber` trait methods:
+//! + [`observe_event`], called when an `Event` takes place,
+//! + [`enter`], called when execution enters a `Span`,
+//! + [`exit`], called when execution exits a `Span`
+//!
+//! In addition, subscribers may implement the [`enabled`] function to _filter_
+//! the notifications they receive based on [metadata] describing each `Span`
+//! or `Event`. If a call to `Subscriber::enabled` returns `false` for a given
+//! set of metadata, that `Subscriber` will *not* be notified about the
+//! corresponding `Span` or `Event`. For performance reasons, if no currently
+//! active subscribers express  interest in a given set of metadata by returning
+//! `true`, then the corresponding `Span` or `Event` will never be constructed.
+//!
+//! # Usage
+//!
+//! First, add this to your `Cargo.toml`:
+//!
+//! ```toml
+//! [dependencies]
+//! tokio-trace = { git = "https://github.com/tokio-rs/tokio" }
+//! ```
+//!
+//! Next, add this to your crate:
+//!
+//! ```rust
+//! #[macro_use]
```

Comment:
Since this is a new crate, should we just start showing how to import with the newer macro importing ability? AKA, `use tokio_trace::span`, etc?

Reply:
The newer macro import syntax doesn't work on Rust 1.26.0, which is tokio's minimum supported Rust version. We could use the new syntax, but then we'd have to stop trying to compile this example as a doctest, as it would break the 1.26.0 CI build.